dali/operators/image/convolution/filter.cc

```diff
@@ -0,0 +1,110 @@
+
+#include "dali/operators/image/convolution/filter.h"
+#include "dali/pipeline/operator/common.h"
+
+namespace dali {
+
+DALI_SCHEMA(experimental__Filter)
+    .DocStr(R"code(Convolves the image with the provided filter.
+
+.. note::
+  In fact, the operator computes a correlation, not a convolution,
+  i.e. the order of filter elements is not flipped when computing the product of
+  the filter and the image.
+
+)code")
+    .NumInput(2, 3)
+    .NumOutput(1)
+    .AllowSequences()
+    .InputDox(0, "data", "TensorList", R"code(Batch of input samples.
+
+Sample can be an image, a video or volumetric (3D) data.
+Samples can contain channels: channel-first and channel-last layouts are supported.
+In case of video/sequences, the frame extent must preced the channels extent, i.e.,
+for example, a video with ``"FCHW"`` layout is supported, but ``"CFHW"`` samples are not.
+
+Samples with the following types are supported:
+int8, int16, uint8, uint16, float16, float32.
+
+Please note that the intermediate type used for the computation is always float32.)code")
+    .InputDox(1, "filter", "TensorList", R"code(Batch of filters.
+
+For inputs with two spatial dimensions (images or video), each filter must be a 2D array
+(or a sequence of 2D arrays to be applied
+:func:`per-frame<nvidia.dali.fn.per_frame>` to a video input).
+For volumetric inputs, the filter must be a 3D array.
+The filter values must have float32 type.)code")
+    .InputDox(2, "fill_value", "TensorList", R"code(Batch of scalars used for padding.
+
+If ``"border"`` is set to ``"constant"``, the input samples will be padded with
+the corresponding scalars when convolved with the filter.
+The scalars must be of the same type as the input samples.
+For video/sequence input, an array of scalars can be specified to be applied
+:func:`per-frame<nvidia.dali.fn.per_frame>`.)code")
+    .AddOptionalArg("anchor",
+                    R"code(Specifies the position of the filter over the input.
+
+If the filter size is ``(r, s)`` and the anchor is ``(a, b)``, the output
+at position ``(x, y)`` is a product of the filter and the input rectangle spanned between the
+corners: top-left ``(x - a, y - b)`` and bottom-right ``(x - a + r - 1, x - b + s - 1)``.
+
+If the -1 (the default) is specifed, the middle (rounded down to integer)
+of the filter extents is used, which, for odd sized filters, results in the filter
+centered over the input.
+
+The anchor must be, depending on the input dimensionality, a 2D or 3D point whose each extent lies
+within filter boundaries (``[0, ..., filter_extent - 1]``). The ordering of anchor's extents
+corresponds to the order of filter's extents.
+
+The parameter is ignored in ``"valid"`` mode.
+.)code",
+                    std::vector<int>{-1}, true, true)
+    .AddOptionalArg("border",
+                    R"code(Controls how to handle out-of-bound filter positions over the sample.
+
+Supported values are: ``"reflect_101"``, ``"reflect_1001"``, ``"wrap"``,
```

Comment:
In the tests I've seen just "101" and "1001" values and not "reflect_*". Is it something different?

Reply:
It's the same, just out of convenience. 

The boundary.h

```
if (type == "reflect_1001" || type == "reflect1001" || type == "1001")
```

Reply:
Ok, understood